new-runtimepoc/Pester.Runtime.psm1

```diff
@@ -597,9 +597,9 @@ function Invoke-TestItem {
                 -NoNewScope
 
             $Test.FrameworkData.Runtime.ExecutionStep = 'Finished'
-            $Test.Passed = $result.Success
+            $Test.ErrorRecord += ($result.ErrorRecord | Where-Object { $_.FullyQualifiedErrorId -notlike '*PesterAssertionFailed*' })
+            $Test.Passed = $Test.ErrorRecord.Count -eq 0
```

Comment:
Imho the `result` should determine whether or not the test was successful. And this piece of code should just copy the values from the result to the `Test` structure.

Reply:
Within the `Invoke-ScriptBlock` function, the Success property is set based on the ErrorRecord count from the recursive invocation of the *wrapperScriptBlock* object. The issue is that that ErrorRecord property is only being filled when the test throw an exception, which is not the only valid case anymore, now assertions will add an existing ErrorRecord to the current test without throwing an exception, unless `-ErrorAction Stop` is specified

Reply:
Then the determination of success should move somewhere where it still makes sense and can be done in one step. Ideally. I am not sure what the possibilites are here :)

Reply:
Uhm, I am not sure how to proceed here. We can either extract it to a separate function now, or wait until a more general use case appears